content/docs/rfcs/17/README.md

```diff
@@ -2,127 +2,138 @@
 slug: 17
 title: 17/WAKU-RLN-RELAY
 name: Waku v2 RLN Relay
-status: raw
+status: draft
 tags: waku-core
 editor: Sanaz Taheri <sanaz@status.im>
 ---
 
-The current specification embodies the details of the spam-protected version of `relay` protocol empowered by Rate Limiting Nullifiers (RLN). 
-<!-- More details on RLN can be found in [this spec]() (TODO: to link the spec). -->
+The `17/WAKU-RLN-RELAY` protocol is an extension of `11/WAKU-RELAY` which additionally provides spam protection using [Rate Limiting Nullifiers (RLN)](/spec/32). 
 
-The security objective is to control the number of PubSub messages that each peer can publish per epoch where epoch is a system design parameter, regardless of the published topic.
+The security objective is to contain spam activity in a GossipSub network by enforcing a global messaging rate to all the peers.
+Peers that violate the messaging rate are considered spammers and their message is considered spam.
+Spammers are also financially punished and removed from the system. 
 
 
 **Protocol identifier***: `/vac/waku/waku-rln-relay/2.0.0-alpha1`
 
 # Motivation
 
-In open p2p messaging networks, one big problem is spam-resistance. 
-Existing solutions, such as Whisper’s proof of work, are insufficient, especially for heterogeneous nodes. 
+In open and anonymous p2p messaging networks, one big problem is spam resistance. 
+Existing solutions, such as Whisper’s proof of work are computationally expensive hence not suitable for resource-limited nodes. 
 Other reputation-based approaches might not be desirable, due to issues around arbitrary exclusion and privacy.
 
-We augment the `relay` protocol with a novel, light, and effective spam prevention mechanism which also suits the resource-constrained nodes.
-
-<!-- TODO: Fill in more -->
+We augment the [`11/WAKU2-RELAY`](/spec/11) protocol with a novel construct of [RLN](/spec/32) to enable an efficient economic spam prevention mechanism that can be run in resource-constrained environments.
 
 
 # Flow
-## SetUp and Registration
-A peer willing to publish a message is required to register. 
-Registration is moderated through a smart contract deployed on the Ethereum blockchain. 
-The state of the contract contains the list of registered members. 
-An overview of registration is illustrated in Figure 1.
+The messaging rate is hardcoded to `1` message per `epoch` per definition. 
+The `epoch` is measured as `epoch` = $\lceil$ current Unix epoch time in seconds / `Epoch_Length` $\rceil$ where `Epoch_Length` indicates the length of the `epoch` in seconds. 
+In other words, the `epoch` value gets incremented after every  `Epoch_Length` passed from the Unix epoch event.
+The messaging rate `1` is a fixed value whereas the epoch duration  `Epoch_Length`  is  application-dependent.
```

Comment:
```suggestion
The messaging rate is hardcoded to `1` message per `epoch` per definition. 
The `epoch` is measured as `epoch` = $\lceil$ current Unix epoch time in seconds / `Epoch_Length` $\rceil$ where `Epoch_Length` indicates the length of the `epoch` in seconds. 
In other words, the `epoch` value gets incremented after every  `Epoch_Length` passed from the Unix epoch event.
The messaging rate `1` is a fixed value whereas the epoch duration  `Epoch_Length`  is  application-dependent.
```

The messaging rate is defined by the `period` which indicates how many messages can be sent in a given period.

We define an `epoch` as `lceil(unix_time / period)`. For example, if `unix_time` is 1644810116 and we set `period` to 30, then `epoch` is  `lceil(unix_time/period) = 54827003`.

Note that epoch refers to epoch in RLN and not Unix epoch. This means a message can only be sent every period, where period is up to the application.

See see section [Recommended System Parameters](#recommended-system-parameters) for some recommended ways to set a sensible `period` value depending on the application.

Reply:
thanks, done https://github.com/vacp2p/rfc/pull/484/commits/8954039fe162421351bcacf73cba41ef864bca21